pkg/operator/mirror.go

```diff
@@ -325,13 +326,21 @@ func pinImages(ctx context.Context, dc *declcfg.DeclarativeConfig, resolverConfi
 	for i, b := range dc.Bundles {
 		if !isImagePinned(b.Image) {
 			if dc.Bundles[i].Image, err = image.ResolveToPin(ctx, resolver, b.Image); err != nil {
+				if errors.As(err, &reference.ErrObjectRequired) {
```

Comment:
It would be more efficient to just check if `b.Image == ""`, ex
```go
if b.Image == "" {
    logrus.Warnf("bundle %s: bundle image not set", b.Name)
    continue
}
if !isImagePinned(b.Image) {
   ...
```